pinc/page_controls.inc

```diff
@@ -0,0 +1,152 @@
+            $project = new Project($projectid); // throws NonexistentProjectException
+
+            $result = DPDatabase::query("SELECT image FROM {$projectid} ORDER BY image ASC");
+            if(0 == $result->num_rows)
+            {
+                throw new UnexpectedValueException(sprintf(_("There are no pages in %s"), $projectid));
+            }
+            while($row = mysqli_fetch_assoc($result))
+            {
+                $images[] = $row["image"];
+            }
+            mysqli_free_result($result);
+
+            if($page !== "")
+            {
+                // find a match for the page
+                $esc_page = DPDatabase::escape($page);
+                $result = DPDatabase::query("SELECT image FROM $projectid WHERE image LIKE '%$esc_page%'");
+                if (mysqli_num_rows($result) !== 0)
+                {
+                    $page = mysqli_fetch_assoc($result)["image"];
+                }
+                else
+                {
+                    $page = "";
+                }
+                mysqli_free_result($result);
+            }
+            if($page === "")
+            {
+                // no data given or couldn't find a match
+                $page = $images[0];
+            }
+            global $code_url;
+            $heading = html_safe($project->nameofwork) . "<a href='$code_url/project.php?id=$projectid'>" . _("Go to Project Page") . "</a>";
+
+            if ($round_id == "OCR")
+            {
+                $text_column_name = 'master_text';
+            }
+            else
+            {
+                $text_column_name = get_Round_for_round_id($round_id)->text_column_name;
+            }
+            $esc_page = DPDatabase::escape($page);
+            $result = DPDatabase::query("SELECT $text_column_name FROM $projectid WHERE image = '$esc_page'");
+            $row = mysqli_fetch_assoc($result);
+            $text = $row[$text_column_name];
+        }
+        catch(Exception $exception)
+        {
+            $error_message = html_safe($exception->getMessage());
+            $projectid = "";
+        }
+    }
+
+    $page_browser_data = [
+        "projectid" => $projectid,
+        "pages" => $images,
+        "currentPage" => $page,
+        "currentRound" => $round_id,
+        "text" => $text,
+        "heading" => $heading,
+        "errorMessage" => $error_message,
+    ];
+    $page_browser_data = json_encode($page_browser_data);
+
+    return "var pageBrowserData = $page_browser_data;";
+}
+
+function get_mode()
+{
+    $mode = get_enumerated_param($_GET, 'mode', 'image', ['image', 'text', 'imageText']);
+    return "var mode = '$mode';";
+}
+
+function expanded_rounds()
+{
+    global $Round_for_round_id_;
+
+    $expanded_rounds = array_keys($Round_for_round_id_);
+    array_unshift($expanded_rounds, 'OCR');
+    return $expanded_rounds;
+}
+
+function get_dp_data()
```

Comment:
We should be more specific than `dp_data`. `get_proofreading_interface_data_js()`?